superflore/generators/bitbake/gen_packages.py

```diff
@@ -49,22 +48,32 @@ def regenerate_pkg(
     component_name = yoctoRecipe.convert_to_oe_name(
         distro.release_packages[pkg].repository_name)
     recipe = yoctoRecipe.convert_to_oe_name(pkg)
-    # check for an existing recipe
-    prefix = '{0}/meta-ros{1}-{2}/generated-recipes/{3}/{4}'.format(
-        repo_dir,
+    # check for an existing recipe which was removed by clean_ros_recipe_dirs
+    prefix = 'meta-ros{0}-{1}/generated-recipes/*/{2}_*.bb'.format(
         yoctoRecipe._get_ros_version(distro.name),
         distro.name,
-        component_name,
         recipe,
     )
-    existing = glob.glob('{}_*.bb'.format(prefix))
+    existing = overlay.repo.git.status('--porcelain', '--', prefix)
+    if existing:
+        # The git status --porcelain output will look like this:
+        # D  meta-ros2-eloquent/generated-recipes/variants/ros-base_0.8.3-1.bb
+        # we want just the path with filename
+        existing = existing.split()[1]
+    if not existing:
+        # If it isn't shown in git status, it could still exist as normal
+        # unchanged file when --only option is being used
+        import glob
+        existing = glob.glob('{}/{}'.format(repo_dir, prefix))
+        if existing:
```

Comment:
What should happen if the glob returns more than one?

Reply:
> What should happen if the glob returns more than one?

The same as before with pcl-conversions, but it's a bit unclear how --only should behave as it isn't implemented yet. Now it will remove only the first one found, but on the other hand, since this change there shouldn't be meta-ros with multiple recipes for the same component, so using --only on correctly created meta-ros shouldn't find 2nd recipe, I'll add an error when there are 2 and warning in the git status above (so that pcl-conversions trigger just an warning).